server/src/main/proto/spine/server/entity/entity.proto

```diff
@@ -72,3 +75,47 @@ message LifecycleFlags {
     //
     bool deleted = 2;
 }
+
+// Custom fields to store in the storage.
+//
+// These fields should be searchable in an effiefient way.
```

Comment:
This comment is not enough. 

Let's expand it by explaining why we store the fields in this way (e.g. per type etc).